modules/llama_cpp_plugin/src/plugin.cpp

```diff
@@ -0,0 +1,107 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "plugin.hpp"
+
+#include <openvino/runtime/properties.hpp>
+
+#include "compiled_model.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/runtime/internal_properties.hpp"
+#include "openvino/util/log.hpp"
+
+namespace {
+static constexpr const char* wait_executor_name = "LlamaCppWaitExecutor";
+static constexpr const char* stream_executor_name = "LlamaCppStreamsExecutor";
+static constexpr const char* template_exclusive_executor = "LlamaCppExecutor";
+}  // namespace
+
+namespace ov {
+namespace llama_cpp_plugin {
+LlamaCppPlugin::LlamaCppPlugin() : IPlugin() {
+    set_device_name("LLAMA_CPP");
+}
+std::shared_ptr<ov::ICompiledModel> LlamaCppPlugin::compile_model(const std::shared_ptr<const ov::Model>& model,
+                                                                  const ov::AnyMap& properties) const {
+    OPENVINO_DEBUG << "llama_cpp_plugin: LlamaCppPlugin::compile_model" << std::endl;
+    return compile_model(model, properties, {});
+}
+
+std::shared_ptr<ov::ICompiledModel> LlamaCppPlugin::compile_model(const std::string& fname,
+                                                                  const ov::AnyMap& properties) const {
+    return std::make_shared<LlamaCppModel>(fname, shared_from_this());
+}
+std::shared_ptr<ov::ICompiledModel> LlamaCppPlugin::compile_model(const std::shared_ptr<const ov::Model>& model,
+                                                                  const ov::AnyMap& properties,
+                                                                  const ov::SoPtr<ov::IRemoteContext>& context) const {
+    OPENVINO_DEBUG << "llama_cpp_plugin: compile_model called in C++" << std::endl;
+    return std::make_shared<LlamaCppModel>(model->clone(),
+                                           shared_from_this(),
+                                           context,
+                                           get_executor_manager()->get_executor(template_exclusive_executor));
+}
+
+void LlamaCppPlugin::set_property(const ov::AnyMap& properties) {
+    for (const auto& map_entry : properties) {
+        if (map_entry.first == ov::cache_dir.name()) {
+            m_cache_dir = map_entry.second.as<std::string>();
```

Comment:
do we need to support this property? Looks like does not have meaningful implementation in current plugin

Reply:
Removed